src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,300 @@
+        except ClientError as e:
+            logger.error("Failed RunInstances request: %s", e.response.get("ResponseMetadata").get("RequestId"))
+            raise e
+
+
+class Ec2FleetManager(FleetManager):
+    """Manager to create EC2 instances fleet using create_fleet API."""
+
+    def _evaluate_launch_params(self, count, launch_overrides):
+        """Evaluate parameters to be passed to create_fleet call."""
+        overrides = []
+        for instance_type in self._compute_resource_config.get("InstanceTypeList", []):
+            overrides.append({"InstanceType": instance_type["InstanceType"]})
+
+        launch_params = {
+            "LaunchTemplateConfigs": [
+                {
+                    "LaunchTemplateSpecification": {
+                        # LaunchTemplate is different for every compute resources in every queue
+                        "LaunchTemplateName": f"{self._cluster_name}-{self._queue}-{self._compute_resource}",
+                        "Version": "$Latest",
+                    },
+                    "Overrides": overrides,
+                }
+            ],
+            "TargetCapacitySpecification": {
+                "TotalTargetCapacity": count,
+                "DefaultTargetCapacityType": "on-demand"
+                if self._queue_config["CapacityType"] == "ONDEMAND"
+                else "spot",
+            },
+            "Type": "instant",
+            "SpotOptions": {
+                "AllocationStrategy": self._queue_config["AllocationStrategy"],
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+            "OnDemandOptions": {
+                "AllocationStrategy": self._queue_config["AllocationStrategy"],
+                "CapacityReservationOptions": {"UsageStrategy": "use-capacity-reservations-first"},
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+        }
+
+        launch_params.update(launch_overrides)
+        if launch_overrides:
+            logger.info("Found CreateFleet parameters override. Launching instances with: %s", launch_params)
+        return launch_params
+
+    def _launch_instances(self, launch_params):
+        """Launch a batch of ec2 instances."""
+        try:
+            response = create_fleet(self._region, self._boto3_config, launch_params)
+            for error in response.get("Errors", []):
+                logger.error("Error in CreateFleet request: %s - %s", error.get("ErrorCode"), error.get("ErrorMessage"))
```

Comment:
is an error in the fleet response also always an error for our process? or is an error returned also when a subset of the instance types is not available and we don't want to mark it as an error?

Reply:
This is something still to be verified/tested, from boto3 create_fleet documentation it is not clear.
This is a behaviour different from the standard boto3 calls on which a ClientError exception is raised.